openedx/features/course_experience/url_helpers.py

```diff
@@ -159,18 +171,33 @@ def make_learning_mfe_courseware_url(
     strings. They're only ever used to concatenate a URL string.
     `params` is an optional QueryDict object (e.g. request.GET)
     """
-    mfe_link = f'{settings.LEARNING_MICROFRONTEND_URL}/course/{course_key}'
+    mfe_link = f'/course/{course_key}'
+    get_params = params.copy() if params else None
+    query_string = ''
+
+    if preview:
+        if len(get_params.keys()) > 1:
+            get_params.pop('preview')
+        else:
+            get_params = None
+
+        if (unit_key or sequence_key):
+            mfe_link = f'/preview/course/{course_key}'
 
     if sequence_key:
         mfe_link += f'/{sequence_key}'
 
         if unit_key:
             mfe_link += f'/{unit_key}'
 
-    if params:
-        mfe_link += f'?{params.urlencode()}'
+    if get_params:
+        query_string = get_params.urlencode()
 
-    return mfe_link
+    url_parts = list(urlparse(settings.LEARNING_MICROFRONTEND_URL))
+    url_parts[2] = mfe_link
+    url_parts[4] = query_string
+
+    return urlunparse(url_parts)
```

Comment:
@KristinAoki This code assumes that the Learning MFE is hosted at the root of a domain, which is true for edx.org but not true for anyone who deploys Tutor. It breaks all `jump_to` links in Tutor, by sending them to:

* `/course/{course_key}/...` and
* `/preview/course/{course_key}/...`

instead of:


* `/learning/course/{course_key}/...` and
* `/learning/preview/course/{course_key}/...`

Could you modify this so that it works regardless of which path the Learning MFE is based on?

Reply:
I am assuming that it is the url builder that is causing the issues, so I will revert back to the string builder for the returned path.